File: examples/renewal/output.py
from dataclasses import dataclass

import numpy as np
from numpy.typing import NDArray


@dataclass
class RenewalOutput:
    infection_incidence: NDArray[np.uint64]
    symptomatic_incidence: NDArray[np.uint64]

    @staticmethod
    def new(length: int) -> "RenewalOutput":
        return RenewalOutput(
            infection_incidence=np.zeros(length, dtype=np.uint64),
            symptomatic_incidence=np.zeros(length, dtype=np.uint64),
        )

File: examples/renewal-rs/src/output.rs
#[derive(Default)]
pub struct RenewalOutput {
    pub infection_incidence: Vec<u64>,
    pub symptomatic_incidence: Vec<u64>,
}

impl RenewalOutput {
    pub fn new(len: usize) -> RenewalOutput {
        RenewalOutput {
            infection_incidence: vec![0; len],
            symptomatic_incidence: vec![0; len],
        }
    }
}
